lib/redix.ex

```diff
@@ -148,38 +155,20 @@ defmodule Redix do
       iex> Redix.start_link(host: "example.com", port: 9999, password: "secret")
       {:ok, #PID<...>}
 
-      iex> Redix.start_link(database: 3, name: :redix_3)
+      iex> Redix.start_link([database: 3], [name: :redix_3])
       {:ok, #PID<...>}
 
   """
-  @spec start_link(Keyword.t | binary) :: GenServer.on_start
-  def start_link(uri_or_opts \\ [])
-
-  def start_link(uri) when is_binary(uri) do
-    uri |> Redix.URI.opts_from_uri |> start_link
-  end
+  @spec start_link(binary | Keyword.t, Keyword.t) :: GenServer.on_start
+  def start_link(uri_or_redis_opts \\ [], connection_opts \\ [])
 
-  def start_link(opts) do
-    opts = Keyword.merge(@default_opts, opts)
-    Connection.start_link(Redix.Connection, opts, opts)
+  def start_link(uri, connection_opts) when is_binary(uri) and is_list(connection_opts) do
+    uri |> Redix.URI.opts_from_uri |> start_link(connection_opts)
   end
 
-  @doc """
-  Works like `start_link/1` but accepts a Redis URI *and* a list of options.
-
-  The options you can pass are the ones not about the connection (e.g.,
-  `:socket_opts` or `:name`). Read the docs for `start_link/1` for more
-  information.
-
-  ## Examples
-
-      iex> Redix.start_link("redis://foo.com/2", name: :redix_foo)
-      {:ok, #PID<...>}
-
-  """
-  @spec start_link(binary, Keyword.t) :: GenServer.on_start
-  def start_link(uri, opts) when is_binary(uri) and is_list(opts) do
-    uri |> Redix.URI.opts_from_uri |> Keyword.merge(opts) |> start_link
+  def start_link(redis_opts, connection_opts) when is_list(redis_opts) and is_list(connection_opts) do
+    redis_opts = Keyword.merge(@default_opts, redis_opts)
```

Comment:
Current `@default_opts` is not only about redis but connection also, seems we just need two: `@redis_defaults` and `@conn_defaults`.